### PR TITLE
switch to dune

### PR DESCRIPTION
**Pull request by redianthus**
Hi,

I switched the project to dune.

Cheers.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,50 +0,0 @@
...
-
-.PHONY: build doc test all install uninstall reinstall clean distclean configure
-
-# OASIS_STOP
-
-VERSION=$(shell oasis query version)
-NAME=ocaml-gperftools-$(VERSION)
-
-.PHONY: release
-release:
```
Comment (ygrek) [inline Makefile]: I need this target to stay to actually make releases :penguin: and in general the whole Makefile for developer convenience

File: _oasis [hunk 2]
```diff
@@ -1,45 +0,0 @@
...
-
-Flag strict
-  Description: Strict compile-time checks
-  Default: true
-
-Library gperftools
-  Path: src
-  Modules: Gperftools
-  if flag(strict) && ccomp_type(cc)
-    CCOpt: -std=c89 -pedantic -Wno-long-long -Wextra
```
Comment (ygrek) [inline _oasis:27]: it is sad to see these options gone

File: src/dune [hunk 3]
```diff
@@ -0,0 +1,5 @@
+(library
```
Comment (ygrek) [inline src/dune:1]: it needs public_name to be installable iiuc

**Review by ygrek — CHANGES_REQUESTED** [review]

File: _oasis (hunk 2 above)
Comment (redianthus) [reply _oasis:27]: I can make add them permanently to `c_flags` if you want. I'm not sure there's a way to make them optional with dune.

**Comment by redianthus** [comment]
Fixed the `Makefile` and `public_name`, for the third one, I'm not sure what to do, see my answer to your comment.

**Comment by ygrek** [comment]
Thanks!

**Pull request merged**